profiler/src/ProfilerEngine/Datadog.Profiler.Native/BclEventsParser.cpp

```diff
@@ -0,0 +1,467 @@
+            break;
+        case 11: // ResponseHeadersStart
+            // OnResponseHeadersStart(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+        case 12: // ResponseHeadersStop
+             OnResponseHeadersStop(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+        case 13: // ResponseContentStart
+            // OnResponseContentStart(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+        case 14: // ResponseContentStop
+            OnResponseContentStop(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+        case 15: // RequestFailedDetailed
+            // OnRequestFailedDetailed(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+        case 16: // Redirect
+            OnRedirect(timestamp, pActivityId, pRelatedActivityId, pEventData, cbEventData);
+            break;
+        default:
+            break;
+    }
+}
+
+void BclEventsParser::OnRequestStart(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+    // string scheme
+    // string host
+    // int port
+    // string path
+    // ... more fields
+
+    ULONG offset = 0;
+
+    WCHAR* scheme = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+    if (scheme == nullptr)
+    {
+        return;
+    }
+
+    WCHAR* host = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+    if (host == nullptr)
+    {
+        return;
+    }
+
+    int port = 0;
+    if (!EventsParserHelper::Read(port, pEventData, cbEventData, offset))
+    {
+        return;
+    }
+
+    WCHAR* path = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+    if (path == nullptr)
+    {
+        return;
+    }
+
+    std::string url = shared::ToString(shared::WSTRING(scheme)) + std::string("://") + shared::ToString(shared::WSTRING(host));
+    if (port != 0)
+    {
+        url = url + ":" + std::to_string(port) + shared::ToString(shared::WSTRING(path));
+    }
+    else
+    {
+        url = url + shared::ToString(shared::WSTRING(path));
+    }
+    _pNetworkListener->OnRequestStart(timestamp, pActivityId, url);
+}
+
+void BclEventsParser::OnRequestStop(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+    // int statusCode
+
+    ULONG offset = 0;
+    uint32_t statusCode = 0;
+    if (!EventsParserHelper::Read(statusCode, pEventData, cbEventData, offset))
+    {
+        return;
+    }
+
+    _pNetworkListener->OnRequestStop(timestamp, pActivityId, statusCode);
+}
+
+void BclEventsParser::OnRequestFailed(std::chrono::nanoseconds timestamp, LPCGUID pActivityId, LPCGUID pRelatedActivityId, LPCBYTE pEventData, ULONG cbEventData)
+{
+    // string exception message
```

Comment:
same.
Do you mean that this function extract the exception message ?

Reply:
make it explicit that the comments map the event payload